OneCommunityGlobal/HighestGoodNetworkApp
# Shereen Add Tool - Created API call

# Description
Updated Add Tool component. Added API call for passing Add Tool form data to the backend, added form validation, made updates to store the add tool data into redux store.
# (WBS): 4.5.7

## Related PRS (if any):
This frontend PR is related to the [#970](https://github.com/OneCommunityGlobal/HGNRest/pull/970) backend PR.


## How to test:
1. check into current branch
2. do npm install and ... to run this PR locally
3. check into [#970](https://github.com/OneCommunityGlobal/HGNRest/pull/970) backend and start the backend
5. go to http://localhost:3000/BMdashboard/tools/add (log in if you are prompted to do so)
6. Enter data in the form
7. check that after clicking submit button, all the form data is saved successfully.
8. Note that if the tool name is already present in the database, it will give an error message saying, 'Oops!! Tool already exists!'.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/c5106b4f-fa2a-44be-9733-bb6df33d2de3
